x/posts/internal/keeper/invariants.go

```diff
@@ -0,0 +1,130 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/posts/internal/types"
+)
+
+// RegisterInvariants registers all posts invariants
+func RegisterInvariants(ir sdk.InvariantRegistry, keeper Keeper) {
+	ir.RegisterRoute(types.ModuleName, "hash256-post-id",
+		Hash256PostIDInvariant(keeper))
+	ir.RegisterRoute(types.ModuleName, "comments-date",
+		ValidCommentsDateInvariants(keeper))
+	ir.RegisterRoute(types.ModuleName, "post-reactions",
+		ValidPostForReactionsInvariants(keeper))
+	ir.RegisterRoute(types.ModuleName, "post-poll-answers",
+		ValidPollForPollAnswersInvariants(keeper))
+}
+
+//TODO not sure if this should be used or can be deleted
+
+// AllInvariants runs all invariants of the module
+func AllInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		if res, stop := Hash256PostIDInvariant(k)(ctx); stop {
+			return res, stop
+		}
+
+		if res, stop := ValidCommentsDateInvariants(k)(ctx); stop {
+			return res, stop
+		}
+
+		if res, stop := ValidPollForPollAnswersInvariants(k)(ctx); stop {
+			return res, stop
+		}
+
+		if res, stop := ValidPostForReactionsInvariants(k)(ctx); stop {
+			return res, stop
+		}
+
+		return "Every invariant condition is fulfilled correctly", true
+	}
+}
+
+// formatOutputIDs concatenate the ids given into a unique string
+func formatOutputIDs(ids types.PostIDs) (outputIDs string) {
+	for _, id := range ids {
+		outputIDs += id.String() + "\n"
+	}
+	return outputIDs
+}
+
+// Hash256PostIDInvariant checks that the all posts have a SHA256 ID
+func Hash256PostIDInvariant(k Keeper) sdk.Invariant {
```

Comment:
What about changing this to make it more generic and call `post.Validate()` instead to make sure all posts are valid?